checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,360 @@
+
+  /** The EnsuresCalledMethods.value field/element */
+  final ExecutableElement ensuresCalledMethodsValueElement =
+      TreeUtils.getMethod(EnsuresCalledMethods.class, "value", 0, processingEnv);
+
+  /** The EnsuresCalledMethods.methods field/element */
+  final ExecutableElement ensuresCalledMethodsMethodsElement =
+      TreeUtils.getMethod(EnsuresCalledMethods.class, "methods", 0, processingEnv);
+
+  /** The CreatesMustCallFor.List.value field/element. */
+  private final ExecutableElement createsMustCallForListValueElement =
+      TreeUtils.getMethod(CreatesMustCallFor.List.class, "value", 0, processingEnv);
+
+  /** The CreatesMustCallFor.value field/element. */
+  private final ExecutableElement createsMustCallForValueElement =
+      TreeUtils.getMethod(CreatesMustCallFor.class, "value", 0, processingEnv);
+
+  /**
+   * Bidirectional map to preserve temporary variables created for nodes with non-empty @MustCall
+   * annotation and the corresponding nodes.
+   */
+  private BiMap<LocalVariableNode, Tree> tempVarToNode = HashBiMap.create();
+
+  /**
+   * Default constructor matching super. Should be called automatically.
+   *
+   * @param checker the checker associated with this type factory
+   */
+  public ResourceLeakAnnotatedTypeFactory(final BaseTypeChecker checker) {
+    super(checker);
+    this.postInit();
+  }
+
+  @Override
+  protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiers(
+        CalledMethods.class, CalledMethodsBottom.class, CalledMethodsPredicate.class);
+  }
+
+  /**
+   * Creates a @CalledMethods annotation whose values are the given strings.
+   *
+   * @param val the methods that have been called
+   * @return an annotation indicating that the given methods have been called
+   */
+  public AnnotationMirror createCalledMethods(final String... val) {
+    return createAccumulatorAnnotation(Arrays.asList(val));
+  }
+
+  @Override
+  public void postAnalyze(ControlFlowGraph cfg) {
+    MustCallConsistencyAnalyzer mustCallConsistencyAnalyzer =
+        new MustCallConsistencyAnalyzer(this, this.analysis);
+    mustCallConsistencyAnalyzer.analyze(cfg);
+    super.postAnalyze(cfg);
+    tempVarToNode.clear();
+  }
+
+  /**
+   * Use the must-call store to get the must-call value of the resource represented by the given
+   * local variables.
+   *
+   * @param localVarWithTreeSet a set of local variables with their assignment trees, all of which
+   *     represent the same resource
+   * @param mcStore a CFStore produced by the MustCall checker's dataflow analysis. If this is null,
+   *     then the default MustCall type of each variable's class will be used.
+   * @return the list of must-call method names
+   */
+  public @Nullable List<String> getMustCallValue(
+      ImmutableSet<LocalVarWithTree> localVarWithTreeSet, @Nullable CFStore mcStore) {
+    MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory =
```

Comment:
No. See my comment on your similar request in the visitor. In general, it is not permitted to store the type factory of a subchecker, because that would cause it to have a stale visitor state.